### PR TITLE
Upgrade to Lucene 8.5.0-snapshot-c4475920b08

### PR DESCRIPTION
**Pull request by dnhatn**
To give [LUCENE-9228](https://issues.apache.org/jira/browse/LUCENE-9228) more CI cycles.

### CODE REVIEW

**Review by jpountz — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/index/search/SimpleQueryStringQueryParser.java [hunk 1]
```diff
@@ -104,7 +105,7 @@ public void setDefaultOperator(BooleanClause.Occur operator) {
     }
 
     @Override
-    protected Query newTermQuery(Term term) {
+    protected Query newTermQuery(Term term, float boost) {
```
Comment (dnhatn) [inline server/src/main/java/org/elasticsearch/index/search/SimpleQueryStringQueryParser.java]: I adjusted this method due to https://issues.apache.org/jira/browse/SOLR-12238.

**Comment by dnhatn** [comment]
@jpountz Can you please take another look? I've adjusted some changes.

**Review by romseygeek — APPROVED** [review]
LGTM.  We can remove our `XIntervals` implementation now as well, I think, but I'll do that in a followup.

**Review by jpountz — APPROVED** [review]

**Comment by dnhatn** [comment]
@jpountz @romseygeek Thanks for reviews.

**Pull request merged**